pages/agenda/[agendaKey]/[teamUID]/index.tsx

```diff
@@ -1,71 +1,151 @@
+import useFetchRequest from 'hooks/agenda/useFetchRequest';
+import styles from 'styles/agenda/TeamDetail/TeamDetail.module.scss';
 
-export default function TeamDetail({ intraId }: { intraId: string }) {
+export default function TeamDetail() {
   const router = useRouter();
-  console.log(router.query);
+  const { agendaKey } = router.query;
   const { teamUID } = router.query;
+  /**
+   * API GET DATA
+   * 1. intraId
+   * 2. agenda Data
+   * 3. teamDetail Data
+   */
+  const intraId = useUser()?.intraId;
 
-  //테스트용
-  intraId = 'leader';
-  // intraId = 'member1';
-  // intraId = 'notMember';
+  const agendaData = useFetchGet<AgendaDataProps>(`/`, {
+    agenda_key: agendaKey,
+  }).data;
 
-  const [teamDetail, setTeamDetail] = useState<TeamDetailProps>({
-    teamName: '팀이름',
-    teamLeaderIntraId: 'leader',
-    teamStatus: TeamStatus.OPEN, //e
-    teamLocation: AgendaLocation.MIX, //e
-    teamContent: '우리팀이세계최강팀',
-    teamMates: [
+  const { data: teamDetailData, getData: getTeamDetail } =
+    useFetchGet<TeamDetailProps>('/team', {
+      teamKey: teamUID,
+      agenda_key: agendaKey,
+    });
+
+  /**
+   *  useEffect / API GET 얻은 데이터 teamDetail, authority 할당
+   */
+  const [teamDetail, setTeamDetail] = useState<TeamDetailProps | null>(null);
+  const [authority, setAuthority] = useState<Authority>(Authority.NONE);
+
+  useEffect(() => {
+    if (intraId && agendaData && teamDetailData) {
+      const userRole =
+        intraId === agendaData?.agendaHost
+          ? Authority.HOST
+          : intraId === teamDetailData.teamLeaderIntraId
+          ? Authority.LEADER
+          : teamDetailData.teamMates.find((mate) => mate.intraId === intraId)
+          ? Authority.MEMBER
+          : Authority.GEUST;
+      setAuthority(userRole);
+      setTeamDetail(teamDetailData);
+    }
+  }, [intraId, agendaData, teamDetailData]);
+
+  /**
+   * Button Click Fucntions
+   */
+  const sendRequest = useFetchRequest().sendRequest;
+
+  const shareTeamInfo = () => {
+    alert('공유하기');
+  };
+
+  const manageTeamDetail = async (method: 'POST' | 'PATCH', url: string) => {
+    sendRequest(
+      method,
+      url,
+      {},
       {
-        intraId: 'leader',
-        coalition: Coalition.GUN, //e
+        agenda_key: agendaKey,
+        teamKey: teamUID,
       },
-      {
-        intraId: 'member1',
-        coalition: Coalition.SPRING,
+      () => {
+        getTeamDetail();
+        if (url === 'team/cancel') router.push(`/agenda/${agendaKey}`);
       },
+      (err: string) => {
+        console.error(err);
+      }
+    );
+  };
+
+  const submitTeamForm = (target: React.FormEvent<HTMLFormElement>) => {
+    target.preventDefault();
+    const formData = new FormData(target.currentTarget);
+    const strData = JSON.stringify(Object.fromEntries(formData));
+    const data = JSON.parse(strData);
+
+    data.teamIsPrivate = data.teamIsPrivate === 'on' ? true : false;
+    switch (data.teamLocation) {
```

Comment:
이건 이런 식으로 해도 깔끔할 것 같아여
```suggestion
const locations={
    '서울': 'SEOUL', ...
   }
const teamLocation = locations[data.teamLocation]; 
```